examples/src/bin/ieee802154_send_frame.rs

```diff
@@ -26,7 +26,8 @@ fn main() -> ! {
 
     let delay = Delay::new();
 
-    let mut ieee802154 = Ieee802154::new(peripherals.IEEE802154, peripherals.RADIO_CLK);
+    let rcc = RadioClockController::new(peripherals.RADIO_CLK);
+    let mut ieee802154 = Ieee802154::new(peripherals.IEEE802154, rcc.split().ieee802154);
```

Comment:
Just looking at this, do we even want manual access to RCC? Since you already store the clock refcounts in a global static, why do we care about the clock controllers in the first place? What if we removed `RADIO_CLK` altogether?

Reply:
I'd just implement all relevant code for the radio singletons, then only radio users could meddle with any of this. What do you think about that?

Reply:
So you mean remove the `RadioClockController` entirely, implement `ModemClockController` for the `WIFI`, `BT` and `IEEE802154` peripheral structs and keep the PHY clock ref count in place?